spacy/errors.py

```diff
@@ -229,8 +229,9 @@ class Errors(metaclass=ErrorsWithCodes):
             "If you're using a Transformer, make sure to install 'spacy-transformers'. "
             "If you're using a custom component, make sure you've added the "
             "decorator `@Language.component` (for function components) or "
-            "`@Language.factory` (for class components).\n\nAvailable "
-            "factories: {opts}")
+            "`@Language.factory` (for class components). If you installed the model "
+            "during the current Python session, you'll need to restart Python to "
```

Comment:
I'm not sure people working in notebooks are going to understand "restart your runtime" from "restart Python"? But I may not be a very good judge of this...

Reply:
I've added Jupyter notebooks to the error message. Open to suggestions if it's still unclear.

Reply:
"restart your Python interpreter or runtime" (longer is just what we need here!)

Reply:
Sorry, I think we posted at about the same time before and I didn't see your update before I wrote the comment above.

I'm also not sure that people will understand "reload their notebook". Let's see, jupyter says "restart kernel" and colab says "restart runtime".

> "If you're in an interactive session or notebook, you'll need to restart your runtime to ..." ???

Another idea: maybe we can put this in the download CLI message instead. And possibly detect notebooks / interactive sessions? We already have utils to detect jupyter for some other warnings/uses.

Reply:
> Another idea: maybe we can put this in the download CLI message instead. And possibly detect notebooks / interactive sessions? We already have utils to detect jupyter for some other warnings/uses.

Yeah, that might be a better place for it. I'll give it a shot.

Reply:
(These are basically the very hardest kind of error messages to write.)

Reply:
I've given it the ol' college try. Feel free to change it.

Reply:
I think this message needs to be pitched more at "novice notebook user" and we can also make an attempt to identify interactive sessions. Let me see...

Reply:
I've added an extra utility function to detect interactive sessions.